### DISCUSSION TITLE
Overview about errors in ts files

### DISCUSSION THREAD

**Discussion by wol81** [ananthakumaran/tide#419] [Q&A]
If typescript code contains some syntax error, or similar things, from a user perspective, I get a small red error marker at the left side of the line where the error occurs, and the error is marked within the line by underlining it.

Both kinds of information are only visible if the line at hand is visible.

For getting that kind of information for the whole file including the invisible parts, or even the whole project, in my current setup I need to step outside emacs and look into some console where the compilation runs.

My question is:
- is it based on some additional packages, configuration, or similar, possible to get more global information about where errors in the source code are?  For example, Eclipse shows a vertical bar with marker symbols for the whole file at the right side (if, e.g., the bar contains a marker at the top, I know that I need to fix an error near the beginning of the file).
- if something like that doesn't yet exist out of the box, what would (in your opinion) be the starting point to implement something like this oneself?  How could one add that to the Lisp code conceptionally?  I assume, that at least the information where errors are goes through tide.

Before diving more into this than with some search terms (as I did before), I would be thankful for some hints / opinions.  So take this question as attempt to evaluate wether there might be a reasonable way to include something like that into my environment.

**Comment by MarcusRiemer** [comment]
Did you stumble over `M-x tide-project-errors`?

**Comment by josteink** [comment]
If you are using flycheck, they should be registered there as well, so `flycheck-next-error` and `flycheck-previous-error` should let you navigate between all the errors, even if they are not on screen.

**Comment by ananthakumaran** [comment]
We use flycheck package and check your mode line, there are packages like doom-modeline, which makes the flycheck error count very obvious. For total project error count, refer `tide-project-errors` mentioned by Marcus.

![image](https://user-images.githubusercontent.com/149238/107954710-d3a3d480-6fc2-11eb-93df-8ebfa2496d86.png)

**Reply by wol81** [reply]
Thank you very much, these are some keywords to start with.
